cfme/storage/volume.py

```diff
@@ -165,6 +165,49 @@ class VolumeSnapshotView(VolumeView):
     cancel = Button('Cancel')
 
 
+class VolumeAttachInstanceEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+
+
+class AttachInstanceView(VolumeView):
+    @property
+    def is_displayed(self):
+        expected_title = 'Attach Cloud Volume "{name}"'.format(name=self.context['object'].name)
+        return (
+                self.in_volume and
+                self.entities.title.text == expected_title and
+                self.entities.breadcrumb.active_location == expected_title)
+
+    instance = BootstrapSelect('vm_id')
+    mountpoint = Input(name='device_path')
+    attach = Button('Attach')
+    cancel = Button('Cancel')
+
+    entities = View.nested(VolumeAttachInstanceEntities)
+
+
+class VolumeDetachInstanceEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+
+
+class DetachInstanceView(VolumeView):
+    @property
+    def is_displayed(self):
+        expected_title = 'Attach Cloud Volume "{name}"'.format(name=self.context['object'].name)
+        return (
+                self.in_volume and
+                self.entities.title.text == expected_title and
+                self.entities.breadcrumb.active_location == expected_title)
+
+    instance = BootstrapSelect(name='vm_id')
+    detach = Button('Detach')
+    cancel = Button('Cancel')
```

Comment:
Is there a `Reset` in the detach view as well? If so, it should be added as a `Button()` here

Reply:
No Reset button at detach View